### ISSUE TITLE
lua not compiled with codeblock

### ISSUE THREAD

**Issue by newfrenchy83** [wesnoth/wesnoth#6562]
### Game and System Information
 - What version of the game are you running?1.17
 - Where did you download it from? (Steam, Mac App Store, our website, built from source, etc.)built from source
 - If you're using a custom build, what compiler did you use, and what commit did you build at?gcc
 - What OS are you running?windows 7

### Describe the bug
when i tried to complie lua project with codeblock i have error message in wesnoth_lua_config.h 
"error line 80 cassert no such file or directory"

before what lua source rename in .c and upload insubmodules, i haven't any problem but now i must remove <Add option="-include ../../src/wesnoth_lua_config.h" /> to lua.cbp for compile

### To Reproduce

1. First.. run codeblock
2. Then... load wesnoth workplace in codebloks projectfiles
3. Lastly... run compilation

### Savefiles
If applicable, attach savefiles to help explain your problem. (To find savefiles, start Wesnoth, click on the `i` button at the bottom left corner of the title screen, and open the `Saved games` directory.)

### Additional context
Add any other context (compiler logs, console output, etc) about the problem here.

**Comment by soliton-** [comment]
Did you checkout the module with `git submodule update --init --recursive` or similar?

**Comment by newfrenchy83** [comment]
yes, the lua is in modules but i don't know if change from .cpp or .c cause bug or not, and after simply upgrade lua, compilation worked fine

**Comment by Pentarctagon** [comment]
What's the actual error message?

**Comment by newfrenchy83** [comment]
the message is
.\..\..\src\wesnoth_lua_config.h|80|fatal error: cassert: No such file or directory|

and when i rename in .cpp is work fine

**Comment by Pentarctagon** [comment]
What's the build command being run though?

**Comment by newfrenchy83** [comment]
> What's the build command being run though?

I command to rebuild lua project

**Comment by Pentarctagon** [comment]
No, I mean the command being run *on the command line* to compile each source file.

**Comment by Pentarctagon** [comment]
Like
```
gcc -o build/release/modules/lua/lapi.o -c -pipe -Wall -Wextra -pthread -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wtrampolines -Werror -fPIE -fstack-protector-strong -O3 -flto=2 -Wno-old-style-cast -Wno-useless-cast -Wno-stringop-overflow -include src/wesnoth_lua_config.h -x c++ -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DHAVE_MYSQLPP -DHAVE_CONFIG_H -D_FORTIFY_SOURCE=2 -D_X11 -DLOAD_REVISION -DFIFODIR='"/var/run/wesnothd"' -DWESNOTH_PATH='"/usr/local/share/wesnoth"' -DLOCALEDIR='"translations"' -DHAS_RELATIVE_LOCALEDIR -DLUA_USE_POSIX -Ibuild -Isrc -isystem /usr/include/mysql -isystem src/modules/mariadbpp/include -isystem src src/modules/lua/lapi.c
```

**Comment by newfrenchy83** [comment]
i use Codeblock and not commandline

**Comment by Pentarctagon** [comment]
Right, but codeblocks isn't a compiler. Codeblocks calls a compiler program. You can see the additional compiler options it sets for example at https://github.com/wesnoth/wesnoth/blob/c1058215e526dcb439ea492753fc8b560b576621/projectfiles/CodeBlocks/liblua.cbp#L32-L38

It needs to also set the `-x c++` option as well for liblua.

**Comment by newfrenchy83** [comment]
it's work! thank you!

**Comment by Wedge009** [comment]
What's the equivalent requirement for MSVC? I think I'm having a similar difficulty.

**Comment by CelticMinstrel** [comment]
It was discussed on Discord/IRC recently. I think the answer was `/EHsc /TP` (added to the Lua COMPILE_FLAGS line in the CMakeLists.txt).

**Comment by Wedge009** [comment]
I was trying to read through that but couldn't really make any sense of it. I also don't think I have MSVC using CMake, I just have the existing project files I was using previously (before the switch to CMake). Well, I'll try to work it out on my own, then.

**Comment by CelticMinstrel** [comment]
If you have the pre-existing project files, you can either find the proper way to do it in the file's compilation settings, or you can just tack those arguments onto the end of the "other arguments" field.

**Comment by Pentarctagon** [comment]
Should be resolved with 9282828f583f245b7d29c92a1052695e088e54ee

**Comment by Wedge009** [comment]
I finally found the VS option to compile as C++ - I generally prefer to use what's built-in than manually adding the `/TP` switch myself. The Exception Handling switch was already set to `/EHsc` so I didn't need to change that.

Sorry for hijacking the codeblocks discussion.
